tests/translator/test_arn_generator.py

```diff
@@ -0,0 +1,34 @@
+from unittest import TestCase
+from parameterized import parameterized
+from mock import Mock
+
+from samtranslator.translator.arn_generator import ArnGenerator, NoRegionFound
+
+
+class TestArnGenerator(TestCase):
+    def setUp(self):
+        ArnGenerator.class_boto_session = None
+
+    @parameterized.expand(
+        [("us-east-1", "aws"), ("cn-east-1", "aws-cn"), ("us-gov-west-1", "aws-us-gov"), ("US-EAST-1", "aws")]
+    )
+    def test_get_partition_name(self, region, expected):
+        actual = ArnGenerator.get_partition_name(region)
+
+        self.assertEqual(actual, expected)
+
+    def test_get_partition_name_raise_NoRegionFound(self):
```

Comment:
You may need a `@patch("boto3.session.Session.region_name", "")` here

Reply:
Ya. I couldn't remember where I was running into the issues before. Added.